data/frame_json_test.go

```diff
@@ -0,0 +1,111 @@
+package data_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGoldenFrameJSON(t *testing.T) {
+	f := goldenDF()
+	a, err := f.MarshalArrow()
+	require.NoError(t, err)
+
+	b, err := data.FrameToJSON(f, true, true) // json.Marshal(f2)
+	require.NoError(t, err)
+	strF := string(b)
+
+	b, err = data.ArrowBufferToJSON(a)
+	require.NoError(t, err)
+	strA := string(b)
+
+	fmt.Println(`{ "arrow": `)
+	fmt.Println(strA)
+	fmt.Println(`, "slice": `)
+	fmt.Println(strF)
+	fmt.Println(`}`)
+
+	assert.JSONEq(t, strF, strA, "arrow and frames should produce the same json")
+
+	goldenFile := filepath.Join("testdata", "all_types.golden.json")
+	if _, err := os.Stat(goldenFile); os.IsNotExist(err) {
+		_ = ioutil.WriteFile(goldenFile, b, 0600)
+		assert.FailNow(t, "wrote golden file")
+	}
+
+	b, err = ioutil.ReadFile(goldenFile)
+	require.NoError(t, err)
+
+	strG := string(b)
+	assert.JSONEq(t, strF, strG, "saved json must match produced json")
+
+	// assert.Equal(t, 1, 2)
+}
+
+func TestGENERATE(t *testing.T) {
```

Comment:
Any suggestions on a better approach?  I hacked this together, but there must be a better pattern!

Reply:
Not sure what you're trying to do - generate code for test cases?

Reply:
This test actually writes code to the console -- I updated the comment so it is a bit more clear.  I could not figure out how to make genny work since arrow has different function names depending on type also!